staging/src/k8s.io/component-base/logs/kube-log-runner/kube-log-runner_test.go

```diff
@@ -0,0 +1,198 @@
+			outputWanted: "filePath: test.log, enableFlush: true, maxSize: 1500, maxAge: 24h30m0shello\n",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resetFlags()
+			err := flag.CommandLine.Parse(tt.commandLine)
+			if err != nil {
+				t.Fatalf("Failed to parse commandline: %v", err)
+				return
+			}
+			// Capture the log output and compare
+			originalStdout := os.Stdout
+			reader, writer, err := os.Pipe()
+			if err != nil {
+				t.Fatalf("Failed to create pipe: %v", err)
+				return
+			}
+			defer func() {
+				err = reader.Close()
+				if err != nil {
+					t.Fatalf("Failed to close pipe reader: %v", err)
+					return
+				}
+			}()
+
+			os.Stdout = writer
+
+			var buf bytes.Buffer
+
+			err = configureAndRun(logrotation.OpenStub)
+			if err != nil {
+				t.Errorf("Expected no error, got %v", err)
+			}
+
+			err = writer.Close()
+			if err != nil {
+				t.Fatalf("Failed to close pipe writer: %v", err)
+				return
+			}
+			// <-done
+
+			_, err = buf.ReadFrom(reader)
+			if err != nil {
+				t.Fatalf("Failed to read from pipe: %v", err)
+				return
+			}
+			output := buf.String()
+			if output != tt.outputWanted {
+				t.Errorf("Expected '%s' but got '%s'", tt.outputWanted, output)
+			}
+
+			// Restore original os.Stdout
+			os.Stdout = originalStdout
+		})
+	}
+}
+
+func TestNegativeConfigureAndRun(t *testing.T) {
+	tests := []struct {
+		name        string
+		commandLine []string
+		errorWanted string
+	}{
+		{
+			name:        "Empty commandline",
+			commandLine: []string{},
+			errorWanted: "not enough arguments to run",
+		},
+		{
+			name:        "Invalid log-file-size flag",
+			commandLine: []string{"-log-file-size=10MM", "test"},
+			errorWanted: "invalid value \"10MM\" for flag -log-file-size: unable to parse quantity's suffix",
+		},
+		{
+			name:        "Negative log-file-size flag",
+			commandLine: []string{"-log-file-size=-10M", "test"},
+			errorWanted: "log-file-size must be non-negative quantity",
+		},
+		{
+			name:        "Invalid log-file-age flag",
+			commandLine: []string{"-log-file-age=-10h", "test"},
+			errorWanted: "log-file-age must be non-negative",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resetFlags()
+			err := flag.CommandLine.Parse(tt.commandLine)
+			if err != nil {
+				if err.Error() != tt.errorWanted {
+					t.Errorf("Expected error: %v, got: %v", tt.errorWanted, err)
+				}
+				return
+			}
+			err = configureAndRun(logrotation.OpenStub)
+			if err == nil || err.Error() != tt.errorWanted {
+				t.Errorf("Expected error: %v, got: %v", tt.errorWanted, err)
+			}
+		})
+	}
+}
+
+func resetFlags() {
+	// Reinitialize the flag to the default value
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+
+	logFileSize = resource.QuantityValue{}
+
+	// Keep sync with the ones in kube-log-runner
```

Comment:
Let's avoid this "keep sync" burden on future maintainers.

Instead of `resetFlags()`, how about a helper function like this:
```
func captureState() (restore func()) {
     enableFlush = flag.CommandLine.Lookup("enable-flush").Value.String()
     ...
     return func() {
         err := flag.CommandLine.Lookup("enable-flush").Value.Set(enableFlush)
         if err != nil {
            panic(err)
        }
        ...
   }
}
```

Then each test can start with:
```
   defer captureState()()
```

Alternatively, move the construction of the `flag.CommandLine` into a function and call that from `main()` and the tests.

Reply:
will update, thanks